src/controller/SetUpCodePairer.cpp

```diff
@@ -243,6 +252,32 @@ CHIP_ERROR SetUpCodePairer::StopConnectOverSoftAP()
     return CHIP_NO_ERROR;
 }
 
+CHIP_ERROR SetUpCodePairer::StartDiscoverOverWiFiPAF(SetupPayload & payload)
+{
+#if CHIP_DEVICE_CONFIG_ENABLE_WIFIPAF
+    ChipLogProgress(Controller, "Starting commissioning discovery over WiFiPAF");
+
+    VerifyOrReturnError(mCommissioner != nullptr, CHIP_ERROR_INCORRECT_STATE);
+    mWaitingForDiscovery[kWiFiPAFTransport] = true;
+    CHIP_ERROR err = DeviceLayer::ConnectivityMgr().WiFiPAFConnect(payload.discriminator, this, OnWiFiPAFSubscribeComplete,
+                                                                   OnWiFiPAFSubscribeError);
+    if (err != CHIP_NO_ERROR)
+    {
+        ChipLogError(Controller, "Commissioning discovery over WiFiPAF failed, err = %" CHIP_ERROR_FORMAT, err.Format());
+        mWaitingForDiscovery[kWiFiPAFTransport] = false;
+    }
+    return err;
+#else
+    return CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE;
+#endif // CONFIG_NETWORK_LAYER_BLE
+}
+
+CHIP_ERROR SetUpCodePairer::StopConnectOverWiFiPAF()
+{
+    mWaitingForDiscovery[kWiFiPAFTransport] = false;
```

Comment:
We handed a pointer to ourselves to `WiFiPAFConnect()`.  We might be about to be destroyed.  This is a security bug.

Reply:
We just pushed a commit which uses weak_ptr to check if the pointer is valid or not while calling WiFiPAFConnect(). Will this still be a problem if having it?